docs/style-spec/expressions.md

```diff
@@ -1,32 +1,34 @@
 # Style Function Expressions
 
 **Constants:**
-- LN2
-- PI
-- e
+- `[ "ln2" ]`
+- `[ "pi" ]`
+- `[ "e" ]`
 
 **Literal:**
 - JSON string / number / boolean literal
 
 **Property lookup:**
-- Feature property:  `{ "ref": "feature", "key": propertyName, "default": defaultValue}`
-- Map property:  `{"ref": "map",  "key": "zoom"|"pitch"|… }`
-  - Note: expressions that refer to the map zoom level are only evaluated at integer zoomslevels. When the map is at non-integer zoom levels, the expression's value will be approximated using linear or exponential interpolation.
+- Feature property: `[ "data", key_expr ]`
+- Map property:
+  - `[ "zoom" ]` (Note: expressions that refer to the map zoom level are only evaluated at integer zoomslevels. When the map is at non-integer zoom levels, the expression's value will be approximated using linear or exponential interpolation.)
+  - `[ "pitch" ]`
```

Comment:
Why would feature property lookups require a `data` function while `zoom` and `pitch` be usable independently of a `map` function, similar to constants? Is it because the set of map properties is finite and well-defined, in contrast to the set of potential feature properties?